### PR TITLE
Onion architecture implemented and EF Core foundation added

### PR DESCRIPTION
**Pull request by tomat-suppe**
We have restructured the entire folder structure to follow the principle of onion architecture.
We have updated namespaces etc. so that all files can find each other.
We have laid the foundation for an EF Core implementation but still need to fix CheepService and DBFacade classes.

Co-authored-by: Sophie Gabriela <slak@itu.dk> 
Co-authored-by: Jasmin <jsho@itu.dk>
Co-authored-by: Maria <arnv@itu.dk>
Co-authored-by: Henriette <hoes@itu.dk>
Co-authored-by: Laura <llsa@itu.dk>
Co-authored-by: Group 3
Co-authored-by: ChatGPT

### CODE REVIEW

**Review by HenrijetteOest — APPROVED** [review]
looks good

File: src/UserFacade/Chirp.Infrastructure/Chirp.Services/CheepService.cs [hunk 1]
```diff
@@ -42,4 +42,4 @@ private static string UnixTimeStampToDateTimeString(double unixTimeStamp)
         return dateTime.ToString("MM/dd/yy H:mm:ss");
     }
```
Comment (Thyme-Cat) [inline src/UserFacade/Chirp.Infrastructure/Chirp.Services/CheepService.cs:44]: Denne klasse har samme UnixTimeStampToDateTimeString( ) metode som DBFacade, ved ikke hvilken der skal beholdes, eller om begge klasser er mere ens.

**Review by llsa-itu — APPROVED** [review]
lgtm :)

File: src/Database/DBFacade.cs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    public DBFacade()
+    {
+        //if (!Directory.Exists("data"))
+        {
+            Directory.CreateDirectory("data");
+            File.Create(DBFilePath);
+        }
+    }
+
+    /*public void FillDatabase(string FileName)
```
[the remark this answers is not in the document]
Comment (tomat-suppe) [reply src/Database/DBFacade.cs:29]: Removed

File: src/Database/DBFacade.cs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        //{
+            // Add a statement to check what is in the data base already.
+            // This is a ductape solution. It ensure we fill the chirp.db 
+            // file noce, instead on every GET request.
+            /*if (!cheepdataExists)
+            {
+                FillDatabase("data/schema.sql");
+                FillDatabase("data/dump.sql");
+                cheepdataExists = true;
+            }*/
```
[the remark this answers is not in the document]
Comment (tomat-suppe) [reply src/Database/DBFacade.cs:69]: Removed

File: src/Chirp.Core/Cheep.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+namespace Chirp.EFCore;
```
[the remark this answers is not in the document]
Comment (tomat-suppe) [reply src/Chirp.Core/Cheep.cs:1]: Vi skal indeed have ændret navne og placering på vores DTO classes til EFCore. Vi merger ind som det er nu, men holder i mente at det skal fixes!

File: src/UserFacade/Chirp.Infrastructure/Chirp.Services/CheepService.cs (hunk 1 above)
Comment (tomat-suppe) [reply src/UserFacade/Chirp.Infrastructure/Chirp.Services/CheepService.cs:44]: Vi vil udfase DBFacade og kommer derfor nok naturligt til at fjerne dublet metoden fra DBFacade class.

**Comment by tomat-suppe** [comment]
Request will be closed - comments have been removed on a new commit, so new pull request (almost identical) will be made now.

**Pull request closed without merging**